source/Politics.cpp

```diff
@@ -258,6 +258,36 @@ string Politics::Fine(PlayerInfo &player, const Government *gov, int scan, const
 
 		int failedMissions = 0;
 
+		// Illegal passengers can only be detected by planetary security.
+		if(!scan)
+		{
+			int64_t fine = ship->Cargo().IllegalPassengersFine(gov, player);
+			if((fine > maxFine && maxFine >= 0) || fine < 0)
+			{
+				maxFine = fine;
+				reason = " for carrying illegal passengers.";
+
+				for(const Mission &mission : player.Missions())
+				{
+					if(mission.IsFailed(player))
+						continue;
+
+					// IllegalCargoMessage applies to both cargo and passengers.
+					string illegalCargoMessage = mission.IllegalCargoMessage();
+					if(!illegalCargoMessage.empty())
+					{
+						reason = ".\n\t";
+						reason.append(illegalCargoMessage);
+					}
+					// Fail any missions with illegal passengers and "stealth" set.
+					if(mission.IllegalCargoFine() > 0 && mission.Passengers() && mission.FailIfDiscovered())
+					{
+						player.FailMission(mission);
+						++failedMissions;
```

Comment:
Looks like this would count the mission as failed twice if the mission has both illegal passengers and illegal cargo and both are caught.

Reply:
there's an IsFailed check a few lines above in both loops, so no?

Reply:
Ah, I see. Yeah, that'll work.